cmd/mcp_client.go

```diff
@@ -0,0 +1,252 @@
+	} else {
+		// When MCP client mode is disabled, just show what's configured
+		serverNames := []string{}
+		for _, server := range mcpConfig.Servers {
+			serverNames = append(serverNames, server.Name)
+		}
+		for name, server := range mcpConfig.MCPServers {
+			if server.Name != "" {
+				serverNames = append(serverNames, server.Name)
+			} else {
+				serverNames = append(serverNames, name)
+			}
+		}
+
+		summary := fmt.Sprintf("Found %d configured MCP server(s): %s (MCP client mode disabled - use --mcp-client to enable)", totalServers, strings.Join(serverNames, ", "))
+		blocks = append(blocks, ui.NewAgentTextBlock().WithText(summary))
+	}
+
+	// Create a map of connected server names for quick lookup
+	connectedServerNames := make(map[string]bool)
+	if mcpClientEnabled {
+		for _, client := range connectedClients {
+			connectedServerNames[client.Name] = true
+		}
+	}
+
+	// Show details for each server with their connection status and tools
+	for _, server := range mcpConfig.Servers {
+		serverBlock := ui.NewAgentTextBlock()
+		serverText := fmt.Sprintf("    • %s (%s)", server.Name, server.Command)
+
+		// Add connection and tools information
+		if mcpClientEnabled {
+			if connectedServerNames[server.Name] {
+				serverText += " - Connected"
+				if tools, exists := serverTools[server.Name]; exists && len(tools) > 0 {
+					toolNames := make([]string, len(tools))
+					for i, tool := range tools {
+						toolNames[i] = tool.Name
+					}
+					serverText += fmt.Sprintf(", Tools: %s", strings.Join(toolNames, ", "))
+				} else {
+					serverText += ", No tools discovered"
+				}
+			} else {
+				serverText += " - Connection failed"
+			}
+		} else {
+			serverText += " - Not connected (--mcp-client disabled)"
+		}
+
+		serverBlock.SetText(serverText)
+		blocks = append(blocks, serverBlock)
+	}
+
+	for name, server := range mcpConfig.MCPServers {
+		serverName := name
+		if server.Name != "" {
+			serverName = server.Name
+		}
+		serverBlock := ui.NewAgentTextBlock()
+		serverText := fmt.Sprintf("    • %s (%s) (legacy)", serverName, server.Command)
+
+		// Add connection and tools information
+		if mcpClientEnabled {
+			if connectedServerNames[serverName] {
+				serverText += " - Connected"
+				if tools, exists := serverTools[serverName]; exists && len(tools) > 0 {
+					toolNames := make([]string, len(tools))
+					for i, tool := range tools {
+						toolNames[i] = tool.Name
+					}
+					serverText += fmt.Sprintf(", Tools: %s", strings.Join(toolNames, ", "))
+				} else {
+					serverText += ", No tools discovered"
+				}
+			} else {
+				serverText += " - Connection failed"
+			}
+		} else {
+			serverText += " - Not connected (--mcp-client disabled)"
+		}
+
+		serverBlock.SetText(serverText)
+		blocks = append(blocks, serverBlock)
+	}
+
+	return blocks, nil
+}
+
+// StartMCPServer starts the MCP server with the given configuration
+func StartMCPServer(ctx context.Context, opt Options) error {
```

Comment:
since we have mcp server and client both in this file, may be keeping the file name `mcp.go` is probably better

Reply:
I think keeping them separate would help with cross-collaboration and provide a clear separation between the MCP client and MCP server, which might otherwise be confusing for new users.